blockchain/test_utils.go

```diff
@@ -0,0 +1,90 @@
+package blockchain
+
+import (
+	"crypto/ecdsa"
+	crand "crypto/rand"
+	"crypto/sha256"
+	mrand "math/rand"
+)
+
+func newHash() Hash {
+	message := make([]byte, 256)
+	crand.Read(message)
+	return sha256.Sum256(message)
+}
+
+func newWallet() Wallet {
+	priv, _ := ecdsa.GenerateKey(curve, crand.Reader)
+	return (*wallet)(priv)
+}
+
+func newTxHashPointer() TxHashPointer {
+	return TxHashPointer{
+		BlockNumber: mrand.Uint32(),
+		Hash:        newHash(),
+	}
+}
+
+func newTxOutput() TxOutput {
```

Comment:
These helpers are great, but perhaps we could expose a parameter generate this `TxOutput` fixture in a way that is not random? Im thinking about testing valid transaction inputs and outputs 🤔